File: app/src/main/java/com/tutv/android/datasource/retrofit/endpoint/GenreAPI.java
package com.tutv.android.datasource.retrofit.endpoint;

import com.tutv.android.domain.Genre;

import java.util.List;

import io.reactivex.Single;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface GenreAPI {
    @GET("genres")
    Single<List<Genre>> getAll();

    @GET("genres/{id}")
    Single<Genre> getById(@Path("id") int genreId, @Query("pagesize") int pagesize, @Query("page") int page);
}

File: app/src/main/java/com/tutv/android/datasource/retrofit/endpoint/GenreAPI.kt
package com.tutv.android.datasource.retrofit.endpoint

import com.tutv.android.datasource.retrofit.annotation.AuthenticatedRequest
import com.tutv.android.domain.Genre
import com.tutv.android.domain.Series
import com.tutv.android.datasource.dto.ResourceViewedDTO
import com.tutv.android.domain.Episode
import com.tutv.android.datasource.dto.SeriesFollowedDTO
import com.tutv.android.datasource.dto.SeriesFollowedResponseDTO
import com.tutv.android.datasource.retrofit.annotation.AvoidForceShowLoginScreenOnAuthFail
import io.reactivex.Single
import retrofit2.http.*

interface GenreAPI {
    @get:GET("genres")
    @get:AuthenticatedRequest
    val all: Single<List<Genre?>?>?

    @GET("genres/{id}")
    fun getById(
        @Path("id") genreId: Int,
        @Query("pagesize") pagesize: Int,
        @Query("page") page: Int
    ): Single<Genre?>?
}